pom.xml

```diff
@@ -88,43 +88,66 @@
           </execution>
         </executions>
       </plugin>
-
-      <plugin>
-        <groupId>org.sonatype.plugins</groupId>
-        <artifactId>nexus-staging-maven-plugin</artifactId>
-        <extensions>true</extensions>
-        <configuration>
-          <serverId>rso</serverId>
-          <nexusUrl>https://repository.sonatype.org/</nexusUrl>
-          <autoReleaseAfterClose>true</autoReleaseAfterClose>
-        </configuration>
-      </plugin>
-
-      <plugin>
-        <artifactId>maven-release-plugin</artifactId>
-        <configuration>
-          <!-- Something in the pom hierarchy overrides the default 'pushChanges == true', so we reset that here. -->
-          <pushChanges>true</pushChanges>
-          <releaseProfiles>gpg-sign</releaseProfiles>
-          <!--
-          CI build runs out of memory during release, with error like this:
-          Failed to execute goal org.apache.maven.plugins:maven-release-plugin:2.5.2:prepare (default-cli) on project nexus-repository-base: Maven execution failed, exit code: '137' -> [Help 1]
-          TODO: Investigate increasing available memory for the CI build
-          -->
-          <arguments>-DskipTests</arguments>
-        </configuration>
-      </plugin>
     </plugins>
   </build>
 
   <profiles>
     <profile>
-      <id>gpg-sign</id>
+      <id>jenkins-release</id>
+      <build>
+        <plugins>
+          <plugin>
+            <groupId>org.sonatype.plugins</groupId>
+            <artifactId>nexus-staging-maven-plugin</artifactId>
+            <version>1.6.8</version>
+            <extensions>true</extensions>
+            <configuration>
+              <autoReleaseAfterClose>true</autoReleaseAfterClose>
+            </configuration>
+          </plugin>
+          <plugin>
+            <artifactId>maven-release-plugin</artifactId>
+            <configuration>
+              <!-- Something in the pom hierarchy overrides the default 'pushChanges == true', so we reset that here. -->
+              <pushChanges>true</pushChanges>
+              <!-- Tests will have been run by CI prior to release. -->
+              <arguments>-DskipTests</arguments>
+            </configuration>
+          </plugin>
+        </plugins>
+      </build>
+    </profile>
+    <profile>
+      <id>circle-ci-release</id>
       <!--
       only sign during deploy phase
       -->
       <build>
         <plugins>
+          <plugin>
+            <groupId>org.sonatype.plugins</groupId>
+            <artifactId>nexus-staging-maven-plugin</artifactId>
+            <extensions>true</extensions>
+            <configuration>
+              <serverId>rso</serverId>
+              <nexusUrl>https://repository.sonatype.org/</nexusUrl>
```

Comment:
Ugh. Really don't like the complexity this adds to the community build. How'd it go with the commit to GitHub via ssh key in the Jenkins CI process? We still need `reproducible` build if we continue down this path.